dialogue-core/src/main/java/com/palantir/dialogue/core/DeprecationWarningChannel.java

```diff
@@ -42,23 +43,25 @@
  */
 final class DeprecationWarningChannel implements EndpointChannel {
     private static final Logger log = LoggerFactory.getLogger(DeprecationWarningChannel.class);
+    // Static cache avoids poor interactions with the jaxrs shim and consumers which recreate clients too aggressively.
+    private static final Cache<LoggingRateLimiterKey, Object> loggingRateLimiter =
```

Comment:
I'm a bit worried about adding static state, would lazily constructing a cache per instance help alleviate your concerns while avoiding static state?

Reply:
we already have a cache per instance, no?

Reply:
fwiw logging is all static state, and this is only used for logging

Reply:
currently the cache key is likely to be such that there is a single entry per instance, but it could easily be changed down the line. I guess since the cache is just for logging its fine

Reply:
Sorry, I neglected to tag you into a slack thread -- we;ve got users who are seeing a warning logged for every rpc instead of rate limited because they're using a legacy codepath (used by both the jaxrs shim and older generated dialogue) which creates a new EndpointChannel instance for every call.